### PR TITLE
Use UTF-8 to write the HTML Report

### PR DESCRIPTION
**Pull request by motofy**
Fix HTML Report mojibake (windows etc)
ReportGenerator input filecode UTF-8

### CODE REVIEW

File: src/org/parosproxy/paros/extension/report/ReportGenerator.java [hunk 1]
```diff
@@ -152,7 +153,7 @@ public static File stringToHtml(String inxml, String infilexsl, String outfilena
 			String line;
 	
 			try {
-				br = new BufferedReader(new FileReader(tempOutfilename));
+				br = new BufferedReader(new InputStreamReader(new FileInputStream(tempOutfilename),"UTF-8"));
```
Comment (thc202) [inline src/org/parosproxy/paros/extension/report/ReportGenerator.java]: Could be used `Files.newBufferedReader` and `StandardCharsets.UTF_8` instead, it's easier to read.

**Comment by thc202** [comment]
Could you add a "ZAP comment" to the beginning of the file? [1]

[1] https://github.com/zaproxy/zaproxy/wiki/DevGuidelines

File: src/org/parosproxy/paros/extension/report/ReportGenerator.java (hunk 1 above)
Comment (thc202) [inline src/org/parosproxy/paros/extension/report/ReportGenerator.java]: The writers are also relying on default charset, probably better to change that as well.

**Comment by kingthorin** [comment]
@motofy are you able to address @thc202's feedback?

**Comment by kingthorin** [comment]
Thanks, could you rebase interactive and fixup into a single commit?

**Comment by motofy** [comment]
Thank you.
I did not know in detail, but tried.
To me, it seems necessary to study JAVA and Github and English.
ZAP, had run well so as hope.

Can you help me fix this issue.

**Comment by kingthorin** [comment]
This might help: https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

**Comment by motofy** [comment]
I hope we did it better than last time.

File: src/org/parosproxy/paros/extension/report/ReportGenerator.java [hunk 2]
```diff
@@ -103,7 +104,7 @@ public static File stringToHtml(String inxml, String infilexsl, String outfilena
 			File stylesheet = null;
 			File outfile = null;
 			StringReader inReader = new StringReader(inxml);
-			String tempOutfilename = outfilename + ".temp"; 
+			String tempOutfilename = outfilename + ".temp";
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/report/ReportGenerator.java]: Formatting changes should be done in a separate PR. https://github.com/zaproxy/zaproxy/wiki/DevGuidelines

**Comment by kingthorin** [comment]
Just one minor things,  otherwise it looks good to me.

**Comment by thc202** [comment]
@motofy could you address the previous comment?

**Comment by thc202** [comment]
Changes look good, just missing the fixup of the commits.

**Comment by thc202** [comment]
@motofy could you fixup the commits to proceed with the merge of the pull request?

**Comment by kingthorin** [comment]
@motofy could you fixup the commits to proceed with the merge of the pull request?

https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md#what-we-zap-team-expect-from-you

**Comment by kingthorin** [comment]
Could you set this PR (branch) to allow modification by maintainers?
https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

**Comment by thc202** [comment]
Will fixup/rebase and merge manually.

**Comment by thc202** [comment]
Changes applied, thank you!

@motofy how would you like to be credited?
https://github.com/zaproxy/zap-core-help/wiki/HelpCredits
